src/bindings/generator/src/dbus_typing.py

```diff
@@ -0,0 +1,174 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+from typing import Tuple
+import dasbus.typing as DBusTypes
+
+
+class DBusTypesExtended(DBusTypes.DBusType):
+    def __init__(self) -> None:
+        super().__init__()
+
+        self._basic_type_mapping_reversed = dict()
+        for key, value in self._basic_type_mapping.items():
+            self._basic_type_mapping_reversed[value] = key
+
+    def is_dbus_char_basic_type(self, char: str) -> bool:
+        return char in self._basic_type_mapping_reversed
+
+    def _map_basic_dbus_type_char_to_pytype_string(self, type_char: str) -> str:
+        if type_char not in self._basic_type_mapping_reversed:
+            raise Exception(f"Unknown basic type '{type_char}'")
+
+        return DBusTypes.get_type_name(self._basic_type_mapping_reversed[type_char])
+
+    def parse_dbus_type_string(self, type_string: str) -> str:
+        """ "parse a D-Bus API string to a python type string (e.g. "s(ss)" -> "Tuple[str, List[str, str]]")
+
+        Parameters:
+        type_string: The D-Bus API string to parse
+
+        Returns:
+        str: The parsed python type string
+        """
+        if len(type_string) < 1:
+            return "", 0
+        if len(type_string) == 1:
+            return self._map_basic_dbus_type_char_to_pytype_string(type_string)
+
+        if type_string[0] == "(":
+            return self._map_dbus_struct_to_pytype_string(type_string)[0]
+        elif type_string[0] == "a":
+            return self._map_dbus_array_type_to_pytype_string(type_string)[0]
+        return self._map_dbus_struct_to_pytype_string(f"({type_string})")[0]
+
+    def _map_dbus_array_type_to_pytype_string(
+        self, type_string: str
+    ) -> Tuple[str, int]:
+        """ "Internal function. Parses a D-Bus API array string.
+
+        Parameters:
+        type_string: The D-Bus API string to parse. Must be an array, starting with "a"
+
+        Returns:
+        Tuple: The parsed python type string and the number of characters read from the string
+        """
+        if len(type_string) < 1 or type_string[0] != "a":
+            raise Exception(
+                "Invalid type definition for array detected in '{t}' - No 'a' found!"
+            )
+
+        if type_string[1] == "y":
```

Comment:
if possible, let's make some constants for the letters like 'y', example DBUS_BYTES_TYPE

Reply:
Extracting it from the `dasbus` library now.